cpp/src/arrow/flight/test-util.h

```diff
@@ -0,0 +1,147 @@
+// Licensed to the Apache Software Foundation (ASF) under one
+// or more contributor license agreements.  See the NOTICE file
+// distributed with this work for additional information
+// regarding copyright ownership.  The ASF licenses this file
+// to you under the Apache License, Version 2.0 (the
+// "License"); you may not use this file except in compliance
+// with the License.  You may obtain a copy of the License at
+//
+//   http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing,
+// software distributed under the License is distributed on an
+// "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+// KIND, either express or implied.  See the License for the
+// specific language governing permissions and limitations
+// under the License.
+
+#include <cstdint>
+#include <cstdio>
+#include <cstring>
+#include <iostream>
+#include <memory>
+#include <string>
+#include <vector>
+
+#include <boost/process.hpp>
+
+#include "arrow/ipc/test-common.h"
+#include "arrow/status.h"
+#include "arrow/test-util.h"
+
+#include "arrow/flight/api.h"
+#include "arrow/flight/internal.h"
+
+namespace bp = boost::process;
```

Comment:
I think boost::process is header-only. I have this here as otherwise it's going to be difficult to deal with process spin-up-spin-down on Windows. So having it as a test dependency is probably ok